### PR TITLE
Added an Extra column in the Article table to store the image link to the article

### PR DESCRIPTION
**Pull request by tfnribeiro**
Added logic so this is retrieved with the article info so it can be displayed in the frontend.

The result can be seen in https://github.com/zeeguu/web/pull/323

### CODE REVIEW

File: tools/migrations/add_img_url_to_article.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+ALTER TABLE
+    `zeeguu_test`.`article`
+ADD
+    COLUMN `img_url_id` INT NULL
+AFTER
+    `deleted`,
+ADD
+    UNIQUE INDEX `img_url_id_UNIQUE` (`img_url_id` ASC) VISIBLE;
```
Comment (mircealungu) [inline tools/migrations/add_img_url_to_article.sql:8]: VISIBLE seems to not be supported by the MySQL engine on the server! if you simply drop it, the query executes.

File: tools/migrations/add_img_url_to_article.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+ALTER TABLE
```
Comment (mircealungu) [inline tools/migrations/add_img_url_to_article.sql:1]: let's please prefix the name of the migrations with <year>-<month> as in the other examples in the folder.

**Review by mircealungu — CHANGES_REQUESTED** [review]
most minor change requests Tiago :)

**Comment by mircealungu** [comment]
@tfnribeiro  - i've made the two changes I pointed to above myself, because i really want to deploy this today ;) by Monday we should come back to a feed with images :)

**Comment by tfnribeiro** [comment]
Sounds good! Sorry, I have the Github account on my personal mail and not on the work mail so I didn't get a notification about this.

I wanted to talk about the SQL file names, because technically we could have a PR go through a few changes before being merged, and if we would want to keep track of when it was deployed maybe it would be best to have a run migration scripts that runs a file, and then recreates it with the date?

**Comment by mircealungu** [comment]
Your observation is very sharp... the reason for the date was in my mind more of a general ordering of the SQL changes than a precise date when the query was run... it's true that it's not very clear why the general ordering si needed ... it just feels nicer somehow? :)

**Comment by tfnribeiro** [comment]
I was more thinking about if we ever wanted to "rollback" then we would have sort of an idea of the order the changes were made, but I think it's OK. Next time I will put the date in the name :)

**Comment by mircealungu** [comment]
this is a very good point! 
it would be most useful to keep track of the order of the changes! 
and now I re-read your idea with the running of the migration script and renaming after. 
this is super-cool. I never heard it before! 
you could even run all migrations that don't have a date in alphabetical order and then when you successfully ran one you rename it with the date!

**Comment by mircealungu** [comment]
Or should we discard articles with path longer than 255? 

These are several examples of articles with long path length: 

￼

There are only two more than 255, the last two above.

If we look at articles that are more than 240 we have a few more: 

/lifeandstyle/ng-interactive/2022/nov/22/christmas-gift-guide-2022-australia-good-gifts-ideas-best-xmas-presents-how-to-shop-for-her-him-by-age-men-them-mum-dad-kids-friends-family-top-unique-thoughtful-useful-online-present-novelty-shopping

/lifeandstyle/ng-interactive/2022/nov/25/christmas-gift-guide-2022-australia-good-gifts-ideas-best-xmas-presents-how-to-shop-for-her-him-by-age-men-them-mum-dad-kids-friends-family-top-unique-thoughtful-useful-online-present-novelty-shopping

/lifeandstyle/ng-interactive/2022/nov/29/christmas-gift-guide-2022-australia-good-gifts-ideas-best-xmas-presents-how-to-shop-for-her-him-by-age-men-them-mum-dad-kids-friends-family-top-unique-thoughtful-useful-online-present-novelty-shopping

/lifeandstyle/ng-interactive/2022/dec/02/christmas-gift-guide-2022-australia-good-gifts-ideas-best-xmas-presents-how-to-shop-for-her-him-by-age-men-them-mum-dad-kids-friends-family-top-unique-thoughtful-useful-online-present-novelty-shopping

/lifeandstyle/ng-interactive/2022/dec/06/christmas-gift-guide-2022-australia-good-gifts-ideas-best-xmas-presents-how-to-shop-for-her-him-by-age-men-them-mum-dad-kids-friends-family-top-unique-thoughtful-useful-online-present-novelty-shopping

/nyheter/innenriks/i/69VRgz/hoeyres-dagligvare-talsperson-lene-westgaard-halle-sier-hun-tror-coop-kan-forsvinne-fra-dagligvaremarkedet-vi-liker-ikke-at-det-bare-er-tre-store-aktoerer-innen-dagligvare-men-det-kan-bli-verre-det-kan-bli-to-sier-hun

/indland/art9407245/L%C3%A6serne-fort%C3%A6ller-om-deres-dybeste-familiehemmeligheder-%C2%BBDa-jeg-kom-hjem-tr%C3%A6ngte-min-mor-mig-op-i-en-krog.-Jeg-sagde-at-hvis-hun-slog-mig-ville-jeg-sl%C3%A5-hende-s%C3%A5-h%C3%A5rdt-jeg-kunne.-S%C3%A5-slog-hun%C2%AB

/politique/macron/ces-inconnus-qui-dirigent-la-france-dominique-boutonnat-et-stephane-sitbon-gomez-soft-power-dominique-boutonnat-le-petit-donateur-de-macron-et-grand-argentier-du-cinema#utm_source=RSS-MARIANNE&utm_medium=Flux-Rss&utm_campaign=RSS_general

/societe/agriculture-et-ruralite/mort-de-la-fille-de-l-agricultrice-decedee-en-ariege-victoire-de-trump-dans-le-new-hampshire-frappes-americaines-en-irak-les-3-infos-de-la-nuit#utm_source=RSS-MARIANNE&utm_medium=Flux-Rss&utm_campaign=RSS_general

/societe/agriculture-et-ruralite/la-fnsea-demande-des-aides-immediates-l-assemblee-favorable-a-une-liberte-garantie-sur-l-ivg-flambee-des-actes-antisemites-les-3-infos-de-la-nuit#utm_source=RSS-MARIANNE&utm_medium=Flux-Rss&utm_campaign=RSS_general

The first few examples here are just shitty SEO titles… we would not lose much by not including them. 

Then again, it wouldn’t be a terrible expense to increase to 512 chars.

Do you see any other perspective that I’m missing? 

Ciao ciao,
M.

**Comment by mircealungu** [comment]
So the simplest thing would be to just drop any article with a too long URL at crawl time?
That would be nice, because it would help us decrease the number of exceptions in Sentry.
And I’d love if we could constantly decrease those till it becomes a special situation that we have exceptions :)

Cheers,
M.

Sent with a Spark

**Pull request merged**